### PR TITLE
Added support for private/public key pairs with key identifiers

### PR DESCRIPTION
**Pull request by ivanut**
This can be used instead of using a certificate (virksomhetssertifikat). Updated tests.

### CODE REVIEW

**Comment by ivanut** [comment]
@zapodot or @Torgeir-Hansen, please take a look at my PR.

**Comment by ivanut** [comment]
BTW, this is the announcement for the key pair functionality: https://samarbeid.digdir.no/id-porten/ny-funksjonalitet-i-min-profil/1042

**Comment by jarleborsheim** [comment]
Thank you @ivanut! We're taking a look at it now 👍

File: KS.Fiks.Maskinporten.Client/MaskinportenClientConfiguration.cs [hunk 1]
```diff
@@ -9,14 +11,25 @@ public class MaskinportenClientConfiguration
...
             string issuer,
             int numberOfSecondsLeftBeforeExpire,
-            X509Certificate2 certificate,
+            X509Certificate2 certificate = null,
+            RSA privateKey = null,
+            RSA publicKey = null,
+            string keyIdentifier = null,
             string consumerOrg = null)
         {
+            if (certificate == null && (privateKey == null || publicKey == null))
```
Comment (exoen) [inline KS.Fiks.Maskinporten.Client/MaskinportenClientConfiguration.cs:20]: Maybe add a check if both certificate and keys are provided and throw exception in that case also as one is supposed to provider just one of them

**Review by exoen — APPROVED** [review]
Looks good, thank you!

**Pull request merged**